### ISSUE TITLE
Collect metadata on each program

### ISSUE THREAD

**Issue by briandfoy** [briandfoy/PerlPowerTools#4]
Collect meta data for all the programs. Store them in a consistent way in the program for now.

See the bin/factor program for a start. See the top =pod section

**Comment by svatsan** [comment]
@briandfoy close this issue? Looks like this was already handled via #8. Unless you're looking for something more

**Comment by briandfoy** [comment]
I'd like to fill in the holes in many of the entries.

**Comment by svatsan** [comment]
Fair enough, I can perhaps take a look around sometime next week to see what's missing

**Comment by jgpuckering** [comment]
Brian,

I whipped up a script to extract the metadata and print it as a tab-separated table for review in Excel.  Here's what I found:

```
Missing Authors:  tar, tty

Missing email addresses:
  Brad Appleton       test
  Jeffrey S. Haemer   addbib, asa, clear, colrm, mkfifo, tsort, uname
  Joshua Gross        date
  Louis Krupp         mkfifo
  Michael Mikonos     base64, hexdump, morse
  Paul Grassie        grep
  dkulp               ar, file, time

Non-standard email:
  Nathan Scott Thompson, "quimby at city-net dot com"
  (in deroff, strings)

Missing license:
  arch, dc, expr, hexdump, id, kill, mimedecode, 
  ppt, strings, sum, tar, tty, whois, xargs
```

Some of these holes can be plugged.  For example, I found an email for Brad Appleton in CPAN -- but without verification it might be pointless to add it.  But I'd be happy to go through that process.

Nathan Scott Thompson has a email but it's inconsistent with the rest.  Easy to fix.

Speaking of inconsistencies, the Description for rot13 starts with a propercase word.  All others are lowercase.  I'd like to change it to lowercase.

I didn't actually validate the License info.  I suspect you already did that, but if you'd like an independent review checking the metadata entry against the POD in the file, I can do that.

As for the missing licenses, most show as perl_5 in MetaCPAN -- probably the default.  Not sure whether the files should be updated accordingly.  Most of these are so old that it seems likely they never had any license information.  Is it valid to add it now???  At Cognos and then IBM I used to do compliance work, including some related to copyright.  It's a complex area.  Nonetheless, it seems to me that if the code is published via MetaCPAN and a license is listed on the page, and most scripts include said license, then all should.  What are your thoughts on that?

As for scripts that are missing metadata, I only found two:  base64 and hexdump.  I have an update for those.

I see you've already released 1.035 with my changes.  That's great!  Let me know if you'd like me to tackle any of the above tasks so this ticket can be closed.  After that I may look into the other tickets.

- Gary

**Comment by briandfoy** [comment]
* If someone didn't declare a license, I leave it blank. perl_5 is probably a good choice if it's absent. I emailed most people I find to ask if they had a preference and filled in their answers. If you can carry that forward, go for it.

* For everything else, it sounds like you have a good plan. For those, though, I'd prefer small, targeted pull requests that deal with single issues (say "Fixes licenses") instead of a big change.
